include/tvm/ir.h

```diff
@@ -16,36 +16,581 @@
+  /*! \return the number of outputs of this function */
+  virtual int num_outputs() const = 0;
+};
+
+/*! \brief reference to a function */
+class FunctionRef : public NodeRef {
+ public:
+  TVM_DEFINE_NODE_REF_METHODS(FunctionRef, NodeRef, FunctionBaseNode);
+};
+
+/*!
+ * \brief Call node.
+ */
+class Call : public ExprNode {
+ public:
+  /*! \brief Possible types of calls. */
+  enum CallType : int {
+    /*! \brief Extern "C" function. */
+    Extern = 0,
+    /*! \brief Extern CXX function. */
+    ExternCPlusPlus = 1,
+    /*! \brief Extern "C" without side-effect. */
+    PureExtern = 2,
+    /*! \brief Halide-style call, evaluates func(args). */
+    Halide = 3,
+    /*! \brief Intrinsic functions. */
+    Intrinsic = 4,
+    /*! \brief Intrinsic functions that are pure. */
+    PureIntrinsic = 5
+  };
+  /*! \brief The name of the function/intrinsic. */
+  std::string name;
+  /*! \brief The arguments. */
+  Array<Expr> args;
+  /*! \brief Type of calls. */
+  CallType call_type;
+  /*! \brief The function to be called. */
+  FunctionRef func;
+  /*! \brief The output value index if func's value is a tuple. */
+  int value_index{0};
+
+  void VisitAttrs(AttrVisitor* v) final {
+    v->Visit("dtype", &type);
+    v->Visit("name", &name);
+    v->Visit("args", &args);
+    v->Visit("call_type", &call_type);
+    v->Visit("func", &func);
+    v->Visit("value_index", &value_index);
+  }
+
+  TVM_DLL static Expr make(Type type,
+                           std::string name,
+                           Array<Expr> args,
+                           CallType call_type,
+                           FunctionRef func = FunctionRef(),
+                           int value_index = 0);
+
+  /*! \return Whether call node is pure. */
+  bool is_pure() const {
+    return (call_type == PureExtern ||
+            call_type == PureIntrinsic ||
+            call_type == Halide);
+  }
+
+  /*!
+   * \return Whether call node corresponds to a defined intrinsic.
+   * \param intrin_name The name of the intrinsic.
+   */
+  bool is_intrinsic(const char* intrin_name) const {
+    return
+        ((call_type == Intrinsic ||
+          call_type == PureIntrinsic) &&
+         name == intrin_name);
+  }
+
+  static constexpr const char* _type_key = "Call";
+  TVM_DECLARE_NODE_TYPE_INFO(Call, ExprNode);
+
+  // Build-in intrinsics
+  static constexpr const char* reinterpret = "reinterpret";
+  static constexpr const char* bitwise_and = "bitwise_and";
+  static constexpr const char* bitwise_not = "bitwise_not";
+  static constexpr const char* bitwise_xor = "bitwise_xor";
+  static constexpr const char* bitwise_or = "bitwise_or";
+  static constexpr const char* shift_left = "shift_left";
+  static constexpr const char* shift_right = "shift_right";
+  static constexpr const char* popcount = "popcount";
+  static constexpr const char* likely = "likely";
+  static constexpr const char* glsl_texture_store = "glsl_texture_store";
```

Comment:
Do we use this?

Reply:
we do have a semi-functioning webgl backend that uses this intrinsic